java/cdk/src/main/java/sleeper/cdk/Utils.java

```diff
@@ -191,9 +191,17 @@ public static <T extends InstanceProperties> T loadInstanceProperties(
             new NewInstanceValidator(AmazonS3ClientBuilder.defaultClient(),
                     AmazonDynamoDBClientBuilder.defaultClient()).validate(properties, propertiesFile);
         }
-
+        String previousVersion = properties.get(VERSION);
+        String builtVersion = getVersion();
         SystemDefinedInstanceProperty.getAll().forEach(properties::unset);
-        properties.set(VERSION, getVersion());
+
+        String update = tryGetContext.apply("update");
+        if ("true".equalsIgnoreCase(update)) {
+            if (!previousVersion.equals(builtVersion)) {
+                throw new MismatchedVersionException("Versions do not match, cannot perform a cdk update");
```

Comment:
This exception message could state the local version and the deployed version, and explain that you can either run an upgrade in order to change versions (though that's not implemented yet), or use the same version that's deployed.

It's slightly complicated in that there are multiple ways you could get to this, like through the admin client or running the CDK directly.